core/capabilities/gateway_connector/service_wrapper_test.go

```diff
@@ -0,0 +1,98 @@
+package gatewayconnector
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/chainlink/v2/core/config/toml"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	chainlink "github.com/smartcontractkit/chainlink/v2/core/services/chainlink"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/keys/ethkey"
+	ksmocks "github.com/smartcontractkit/chainlink/v2/core/services/keystore/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/test-go/testify/mock"
+)
+
+func generateConfig(addr common.Address) (chainlink.GeneralConfig, error) {
```

Comment:
You can put more shared logic in this helper. Instead of returning config, return the whole Wrapper object with mocks already set.

Reply:
Ok, I'd wanted to do that but apparently misunderstood your earlier comment.